specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json

```diff
@@ -317,6 +317,103 @@
         }
       }
     },
+    "/{scope}/providers/Microsoft.CostManagement/alerts/{alertId}": {
+      "get": {
+        "tags": [
+          "Alerts"
+        ],
+        "operationId": "Alerts_Get",
+        "description": "Gets the alert for the scope by alert ID.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
+        },
+        "x-ms-examples": {
+          "SubscriptionAlerts": {
+            "$ref": "./examples/SingleSubscriptionAlert.json"
+          },
+          "ResourceGroupAlerts": {
+            "$ref": "./examples/SingleResourceGroupAlert.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/scopeAlertParameter"
+          },
+          {
+            "$ref": "#/parameters/alertIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/Alert"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "patch": {
```

Comment:
Mmm, not sure about using Patch as an action (not just for updating state to signal dismiss state). This is typically done with POST. Calling it out for ARM review.

Reply:
thanks for pointing out the enum issue. I made an update adding all missing enums.

Reply:
I'm really confused about the purpose of the API. The contract looks like a legitimate PATCH operation, but you call it a "dismiss" API.

If you are planning to allow patching of other alert properties in the future, then I think it makes sense to keep PATCH as the verb but you should change all the descriptions. ("Dismiss" in examples is fine, though.)

If dismissing is the only modification that you will allow, then it should be a POST /dismiss on the alert resource.

---
In reply to: [421001556](https://github.com/Azure/azure-rest-api-specs/pull/9329#discussion_r421001556) [](ancestors = 421001556)

Reply:
@majastrz  That's great feedback. Completely agree with you on this one. It should be a /dismiss. However, this is the currently published API which customers are actively using so we are not looking to make any breaking changes to contract here. When we do rev the API version in the near future we will incorporate this.
Do let me know if you see any concerns on this.

Reply:
It's fine to keep it, but you should update the description accordingly (except for examples). That change is swagger-only and has no bearing on your backend.